### PR TITLE
Docs: Port code examples to C# (M, N, O, P, Q, R)

### PR DESCRIPTION
**Pull request by HaSa1002**
Includes:
 * MarginContainer
 * NavigationPolygon
 * Node
 * NodePath
 * OS
 * PackedByteArray
 * PackedScene
 * PacketPeerUDP
 * PCKPacker
 * Performance
 * PhysicsShapeQueryParameters2D
 * PhysicsShapeQueryParameters3D
 * PrimitiveMesh
 * ProjectSettings

### CODE REVIEW

File: doc/classes/OS.xml [hunk 1]
```diff
@@ -118,13 +136,28 @@
...
-				[/codeblock]
+				[/gdscript]
+				[csharp]
+				var arguments = new Godot.Collections.Dictionary();
+				foreach (var argument in OS.GetCmdlineArgs())
+				{
+				    if (argument.Find("=") &gt; -1)
+				    {
+				        string[] keyValue = argument.Split("=");
+				        // We don't have an LStrip StringExtension :( so we use Substring
```
Comment (aaronfranke) [inline doc/classes/OS.xml:154]: Can be fixed here after #43250 is merged. Same with HexEncode.

**Review by aaronfranke — CHANGES_REQUESTED** [review]

File: doc/classes/OS.xml (hunk 1 above)
Comment (HaSa1002) [reply doc/classes/OS.xml:154]: forward-fixed

File: doc/classes/ProjectSettings.xml [hunk 2]
```diff
@@ -937,15 +968,15 @@
...
+			PhysicsServer3D.AreaSetParam(GetViewport().FindWorld().Space, PhysicsServer3D.AreaParameter.Gravity, 9.8);
 			[/codeblock]
 		</member>
 		<member name="physics/3d/default_gravity_vector" type="Vector3" setter="" getter="" default="Vector3( 0, -1, 0 )">
 			The default gravity direction in 3D.
 			[b]Note:[/b] This property is only read when the project starts. To change the default gravity vector at runtime, use the following code sample:
 			[codeblock]
 			# Set the default gravity direction to `Vector3(0, -1, 0)`.
-			PhysicsServer3D.area_set_param(get_viewport().find_world().get_space(), PhysicsServer3D.AREA_PARAM_GRAVITY_VECTOR, Vector3(0, -1, 0))
+			PhysicsServer3D.AreaSetParam(GetViewport().FindWorld().Space, PhysicsServer3D.AreaParameter.GravityVector, Vector3(0, -1, 0))
```
[the remark this answers is not in the document]
Comment (HaSa1002) [reply doc/classes/ProjectSettings.xml:996]: Vector3.Down maybe?

**Comment by HaSa1002** [comment]
applied changes of code review

File: doc/classes/ProjectSettings.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (aaronfranke) [reply doc/classes/ProjectSettings.xml:996]: Yes.

**Comment by HaSa1002** [comment]
applied changes from code review

File: doc/classes/PhysicsShapeQueryParameters3D.xml [hunk 3]
```diff
@@ -31,19 +31,34 @@
...
 
-			# Release the shape when done with physics queries.
-			PhysicsServer3D.free_rid(shape_rid)
-			[/codeblock]
+				# Release the shape when done with physics queries.
+				PhysicsServer3D.free_rid(shape_rid)
+				[/gdscript]
+				[csharp]
+				RID shapeRid = PhysicsServer3D.ShapeCreate(PhysicsServer3D.ShapeType.Sphere);
+				float radius = 2.0;
```
Comment (aaronfranke) [inline doc/classes/PhysicsShapeQueryParameters3D.xml:50]:
```suggestion
				float radius = 2.0f;
```

File: doc/classes/Performance.xml [hunk 4]
```diff
@@ -24,14 +24,44 @@
...
+				    var monitorValue = new Callable(this, nameof(GetMonitorValue));
+
+				    Performance.AddCustomMonitor("MyCategory/MyMonitor", monitorValue); // Adds monitor with name "MyName" to category "MyCategory"
+				    Performance.AddCustomMonitor("MyMonitor", monitorValue); // Adds monitor with name "MyName" to category "Custom"
+				    // Note: "MyCategory/MyMonitor" and "MyMonitor" have same name but different ids so above code is valid.
+
+				    Performance.AddCustomMonitor("Custom/MyMonitor", monitorValue); // Adds monitor with name "MyName" to category "Custom"
+				    // Note: "MyMonitor" and "Custom/MyMonitor" have same name and same category but different ids so above code is valid.
+
+				    Performance.AddCustomMonitor("MyCategoryOne/MyCategoryTwo/MyMonitor", monitorValue); // Adds monitor with name "MyCategoryOne/MyCategoryTwo/MyMonitor" to category "Custom".
```
Comment (aaronfranke) [inline doc/classes/Performance.xml:56]:
```suggestion
				    // Adds monitor with name "MyName" to category "MyCategory".
				    Performance.AddCustomMonitor("MyCategory/MyMonitor", monitorValue);
				    // Adds monitor with name "MyName" to category "Custom".
				    // Note: "MyCategory/MyMonitor" and "MyMonitor" have same name but different ids so the code is valid.
				    Performance.AddCustomMonitor("MyMonitor", monitorValue);

				    // Adds monitor with name "MyName" to category "Custom".
				    // Note: "MyMonitor" and "Custom/MyMonitor" have same name and same category but different ids so the code is valid.
				    Performance.AddCustomMonitor("Custom/MyMonitor", monitorValue); 

				    // Adds monitor with name "MyCategoryOne/MyCategoryTwo/MyMonitor" to category "Custom".
				    Performance.AddCustomMonitor("MyCategoryOne/MyCategoryTwo/MyMonitor", monitorValue);
```

Put comments on their own line, especially when super long.

File: doc/classes/Performance.xml [hunk 5]
```diff
@@ -24,14 +24,44 @@
...
+				    var monitor_value = Callable(self, "get_monitor_value")
+
+				    Performance.add_custom_monitor("MyCategory/MyMonitor", monitor_value) # Adds monitor with name "MyName" to category "MyCategory"
+				    Performance.add_custom_monitor("MyMonitor", monitor_value) # Adds monitor with name "MyName" to category "Custom"
+				    # Note: "MyCategory/MyMonitor" and "MyMonitor" have same name but different ids so above code is valid.
+
+				    Performance.add_custom_monitor("Custom/MyMonitor", monitor_value) # Adds monitor with name "MyName" to category "Custom"
+				    # Note: "MyMonitor" and "Custom/MyMonitor" have same name and same category but different ids so above code is valid.
+
+				    Performance.add_custom_monitor("MyCategoryOne/MyCategoryTwo/MyMonitor", monitor_value) # Adds monitor with name "MyCategoryOne/MyCategoryTwo/MyMonitor" to category "Custom".
```
Comment (aaronfranke) [inline doc/classes/Performance.xml:39]:
```suggestion
				    # Adds monitor with name "MyName" to category "MyCategory".
				    Performance.add_custom_monitor("MyCategory/MyMonitor", monitor_value)

				    # Adds monitor with name "MyName" to category "Custom".
				    # Note: "MyCategory/MyMonitor" and "MyMonitor" have same name but different ids so the code is valid.
				    Performance.add_custom_monitor("MyMonitor", monitor_value)

				    # Adds monitor with name "MyName" to category "Custom".
				    # Note: "MyMonitor" and "Custom/MyMonitor" have same name and same category but different ids so the code is valid.
				    Performance.add_custom_monitor("Custom/MyMonitor", monitor_value)

				    # Adds monitor with name "MyCategoryOne/MyCategoryTwo/MyMonitor" to category "Custom".
				    Performance.add_custom_monitor("MyCategoryOne/MyCategoryTwo/MyMonitor", monitor_value)
```

**Review by aaronfranke — CHANGES_REQUESTED** [review]

File: doc/classes/Performance.xml [hunk 6]
```diff
@@ -24,14 +24,53 @@
...
+
+				    // Adds monitor with name "MyName" to category "MyCategory".
+				    Performance.AddCustomMonitor("MyCategory/MyMonitor", monitorValue);
+				    // Adds monitor with name "MyName" to category "Custom".
+				    // Note: "MyCategory/MyMonitor" and "MyMonitor" have same name but different ids so the code is valid.
+				    Performance.AddCustomMonitor("MyMonitor", monitorValue);
+
+				    // Adds monitor with name "MyName" to category "Custom".
+				    // Note: "MyMonitor" and "Custom/MyMonitor" have same name and same category but different ids so the code is valid.
+				    Performance.AddCustomMonitor("Custom/MyMonitor", monitorValue);
```
Comment (aaronfranke) [inline doc/classes/Performance.xml:62]:
```suggestion
				    Performance.AddCustomMonitor("Custom/MyMonitor", monitorValue);
```

My bad :)

**Review by aaronfranke — CHANGES_REQUESTED** [review]

File: doc/classes/Performance.xml (hunk 6 above)
Comment (HaSa1002) [reply doc/classes/Performance.xml:62]: shift happens :D

Comment (HaSa1002) [reply doc/classes/Performance.xml]: stupid me... Forogot to git add

Comment (HaSa1002) [reply doc/classes/Performance.xml]: ...and wonder why the check still fails...

**Review by aaronfranke — APPROVED** [review]
This looks good, except that some of the example code expects #43250 to be merged.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**